File: Authored-Katas/04-Frequency-Analysis-With-Buckets/Solution.rb
def bucketize(arr)
  hsh = Hash.new(0)
  buckets = [nil]

  arr.each_with_index do |num, idx|
    hsh[num] += 1
    buckets[idx+1] = nil
  end
    
  hsh.each do |k,v|
    if buckets[v].nil?
      buckets[v] = [k]
    else
      buckets[v].push(k)
      buckets[v].sort!
    end
  end

  buckets
end

File: Authored-Katas/04-Frequency-Analysis-With-Buckets/Solution.js
function bucketize(arr) {
  let obj = {};
  let buckets = [null];

  for (let i = 0; i < arr.length; i++) {
    let num = arr[i];
    obj[num] = obj[num] + 1 || 1;
    buckets[i+1] = null;
  }

  for (let key in obj) {
    let val = obj[key];
    key = parseInt(key);

    if (buckets[val] === null) {
      buckets[val] = [key];
    }
    else {
      buckets[val].push(key);
    }
  }

  return buckets;
}
